pulsar-client/src/main/java/org/apache/pulsar/client/impl/conf/ClientConfigurationData.java

```diff
@@ -162,6 +165,24 @@ public class ClientConfigurationData implements Serializable, Cloneable {
     )
     private String tlsTrustCertsFilePath = null;
 
+    @ApiModelProperty(
+            name = "tlsPrivateKey",
+            value = "The private key for the client certificate. Only used when tlsTrustStoreStream is non-null."
+    )
+    private PrivateKey tlsPrivateKey = null;
```

Comment:
would it be possible to renew the key?

what about creating a interface like TLSClientMaterialProvider ?

interface TLSClientMaterialProvider {
     PrivateKey getPrivateKey();
     InputStream openTrustStoreStream();
     Certificate[]  getCertificateChain();
}

we can also provide default implementations that read from a local file (with automatic reload) or from a classpath resource

Reply:
Great question. I was just noticing that the regular configuration does not allow for renewing the key, which is likely a missing feature. I was about to change my implementation to match this class: https://github.com/apache/pulsar/blob/330fcb9787d9f822667f0617b22bcae66b4644e5/pulsar-client/src/main/java/org/apache/pulsar/client/impl/auth/AuthenticationDataTls.java#L69-L85

However, that might not work either.

Reply:
I agreed with Enrico, creating a provider is a good idea, because we have too many TLS configs on the builder.